### PR TITLE
[webOS] Video decoder improvements

### PR DESCRIPTION
**Pull request by sundermann**
## Description

Implements pipeline error handling, end of stream handling and proper [segment event handling](https://gstreamer.freedesktop.org/documentation/additional/design/segments.html?gi-language=c). Segment events allow to replace the current seek call which comes with its own drawbacks (H265 files not resuming, not accurate seeking). Usually a demuxer element would send those events but in our case we plug into the pipeline after the demuxer and that's why we need to send those events so the videosink's time is updated.

## How has this been tested?

OLED77C28LB (webOS 7/22)

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecStarfish.cpp [hunk 1]
```diff
@@ -385,6 +385,9 @@ CDVDVideoCodec::VCReturn CDVDVideoCodecStarfish::GetPicture(VideoPicture* pVideo
   if (m_state == StarfishState::ERROR)
     return VC_ERROR;
 
+  if (m_state == StarfishState::ERROR)
```
Comment (popcornmix) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecStarfish.cpp:388]: I don't think this condition could be true (or you would have returned two line up).
Maybe `StarfishState::EOS` would make more sense?

File: tools/depends/target/webos-userland/Makefile [hunk 2]
```diff
@@ -1,5 +1,6 @@
 include ../../Makefile.include WEBOS-USERLAND-VERSION ../../download-files.include
 DEPS= ../../Makefile.include Makefile WEBOS-USERLAND-VERSION ../../download-files.include
+BASE_URL=https://github.com/webosbrew/webos-userland/archive/refs/tags/0.0.9
```
Comment (wsnipex) [inline tools/depends/target/webos-userland/Makefile:3]: uploaded to mirrors

**Review by wsnipex — APPROVED** [review]
great improvement!

File: tools/depends/target/webos-userland/Makefile (hunk 2 above)
Comment (sundermann) [reply tools/depends/target/webos-userland/Makefile]: Thanks, dropped the temporary URL change commit

**Pull request merged**